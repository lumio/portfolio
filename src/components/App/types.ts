interface AppStateType {
    pending : boolean;
    error : boolean;
    data : any;
}

export { AppStateType };
