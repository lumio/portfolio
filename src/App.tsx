import * as React from 'react';
import * as yaml from 'js-yaml';

class App extends React.Component<any, any> {
  constructor( props : any ) {
    super( props );

    this.state = {
      pending: true,
      error: false,
      data: {},
    };

    this.loadContent();
  }

  loadContent() {
    fetch( './content.yml' )
      .then( ( response ) => response.text() )
      .then( ( data ) => yaml.safeLoad( data ) )
      .then( ( result ) => this.setState( {
        pending: false,
        data: result,
      } ) )
      .catch( ( err ) => {
        // tslint:disable-next-line
        console.error( err );
        this.setState( {
          pending: false,
          error: true,
        } );
      } );
  }

  renderPending() {
    return (
      <div>Pending</div>
    );
  }

  renderError() {
    return (
      <div><strong>error</strong></div>
    );
  }

  renderPortfolio() {
    return (
      <div>
        <h1>Hello world</h1>
        <pre>{ JSON.stringify( this.state.data, null, 2 ) }</pre>
      </div>
    );
  }

  render() {
    if ( this.state.pending ) {
      return this.renderPending();
    }

    if ( this.state.error || !this.state.data ) {
      return this.renderError();
    }

    return this.renderPortfolio();
  }
}

export default App;
