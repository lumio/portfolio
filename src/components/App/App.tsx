import * as React from 'react';
import * as yaml from 'js-yaml';

import { AppStateType } from './types';
import ErrorBoundary from '../ErrorBoundary';
import SectionContainer from '../SectionContainer';

class App extends React.Component<{}, AppStateType> {
  constructor( props : {} ) {
    super( props );

    this.state = {
      pending: true,
      error: false,
      data: {},
    };

    this.loadContent();
  }

  loadContent() {
    fetch( './content.yml' )
      .then( ( response ) => response.text() )
      .then( ( data ) => yaml.safeLoad( data ) )
      .then( ( result ) => this.setState( {
        pending: false,
        data: result,
      } ) )
      .catch( ( err ) => {
        if ( !process || !process.env || process.env.NODE_ENV !== 'test' ) {
          // tslint:disable-next-line
          console.error( err );
        }
        this.setState( {
          pending: false,
          error: true,
        } );
      } );
  }

  renderPending() {
    return (
      <div>Pending</div>
    );
  }

  renderError() {
    return (
      <div><strong>error</strong></div>
    );
  }

  render() {
    if ( this.state.pending ) {
      return this.renderPending();
    }

    if ( this.state.error || !this.state.data ) {
      return this.renderError();
    }

    return (
      <ErrorBoundary>
        <SectionContainer sections={ this.state.data } />
      </ErrorBoundary>
    );
  }
}

export default App;
