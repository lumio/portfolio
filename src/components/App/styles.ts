import styled, { keyframes } from 'styled-components';

const pulsate = keyframes`
  from {
    opacity: .6;
    filter: blur( .05rem );
    transform: scale( .9 );
  }
  to {
    opacity: 1;
    filter: blur( 0 );
    transform: scale( 1.1 );
  }
`;

const AppLoadingStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;

  & > span {
    font-size: 1.8rem;
    animation: ${ pulsate } 1.5s ease infinite alternate;
  }
`;

export { AppLoadingStyles };
