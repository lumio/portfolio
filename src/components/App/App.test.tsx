import * as React from 'react';
import * as renderer from 'react-test-renderer';

import App from './App';

const mockContent = `---
titlescreen:
  title: Portfolio
  subtitle: of Henry Ford
  mail: henry [dot] ford [at] example [dot] com
`;

describe( 'App', () => {
  it( 'renders snapshot on data', ( done ) => {
    // FIXME:
    ( fetch as any ).mockResponse( mockContent );

    const renderedApp = renderer.create( <App /> );
    expect( renderedApp.toJSON() ).toMatchSnapshot();
    setTimeout( () => {
      expect( renderedApp.toJSON() ).toMatchSnapshot();
      done();
    }, 100 );
  } );

  it( 'renders snapshot on error', ( done ) => {
    // FIXME:
    ( fetch as any ).mockReject( new Error( 'some error' ) );

    const renderedApp = renderer.create( <App /> );
    expect( renderedApp.toJSON() ).toMatchSnapshot();
    setTimeout( () => {
      // FIXME: renders error
      expect( renderedApp.toJSON() ).toMatchSnapshot();
      done();
    }, 100 );
  } );
} );
