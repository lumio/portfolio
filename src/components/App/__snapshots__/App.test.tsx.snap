// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot on data 1`] = `
<div>
  Pending
</div>
`;

exports[`App renders snapshot on data 2`] = `
<div>
  <pre>
    {
      "data": {
        "title": "Portfolio",
        "subtitle": "of Henry Ford",
        "mail": "henry [dot] ford [at] example [dot] com"
      }
    }
  </pre>
</div>
`;

exports[`App renders snapshot on error 1`] = `
<div>
  Pending
</div>
`;

exports[`App renders snapshot on error 2`] = `
<div>
  <strong>
    error
  </strong>
</div>
`;
